internal/util/openapi/openapi.go

```diff
@@ -0,0 +1,91 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package openapi
```

Comment:
I think it would be clear and reusable if this is moved to kyaml openapi library.

Reply:
This code has dependencies on the k8s libraries, which we do not want in kyaml. If this proves useful, it could be split out into a separate library that can be used with kyaml.

Reply:
Agree. This is implementing kpt's own initSchema() with building blocks in kyaml. Good to have it here for now.